Automattic/wp-calypso
# Activity Log: fix and test restore persistence

The Activity Log's `restoreProgress` reducer has a persistence bug. It's using the following wrapper combination:
```js
withSchemaValidation ( keyedReducer ( withoutPersistence ( reducer ) ) )
```
`withSchemaValidation` enables persistence and then `withoutPersistence` disables it again, resulting in no persistence.

This PR fixes that by using `withPersistence` instead, making the reducer forward-compatible with changes in #50625.

**How to test:**
1. Checkout the commit with the unit tests. Verify they fail.
2. Checkout the commit with the fix. Verify the tests start passing :+1:
